src/worker/mod.rs

```diff
@@ -187,7 +192,16 @@ impl<S: AsyncBufRead + AsyncWrite + Send + Unpin, E: StdError + 'static + Send>
             .callbacks
             .get(job.kind())
             .ok_or(Failed::BadJobType(job.kind().to_string()))?;
-        handler.run(job).await.map_err(Failed::Application)
+        match handler {
+            Callback::Async(cb) => cb.run(job).await.map_err(Failed::Application),
+            Callback::Sync(cb) => {
+                let cb = Arc::clone(cb);
+                spawn_blocking(move || cb(job))
+                    .await
+                    .expect("joined ok")
```

Comment:
This will fail if the provided `Fn` ever panics. May be worth bubbling up as another kind of error (`Failed`) instead?

Reply:
added a local error variant